### PR TITLE
ts provision

### PR DESCRIPTION
**Pull request by chrisnojima**

### CODE REVIEW

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready

File: shared/provision/paper-key/index.tsx [hunk 1]
```diff
@@ -1,18 +1,22 @@
...
 import * as Styles from '../../styles'
 
 type Props = {
-  onBack: () => void,
-  onSubmit: string => void,
-  hint: string,
-  error: string,
-  waitingForResponse?: ?boolean,
+  onBack: () => void
+  onSubmit: (arg0: string) => void
```
Comment (cjb) [inline shared/provision/paper-key/index.tsx]: s/arg0/paperKey/

File: shared/provision/code-page/index.tsx [hunk 2]
```diff
@@ -14,23 +13,22 @@ const greenBackground = require('../../images/illustrations/bg-provisioning-gree
...
+  error: string
+  currentDeviceAlreadyProvisioned: boolean
+  currentDeviceType: DeviceType
+  currentDeviceName: string
+  otherDeviceName: string
+  otherDeviceType: DeviceType
+  tabOverride?: Tab | null
+  textCode: string
+  onBack: () => void
+  onSubmitTextCode: (arg0: string) => void
```
Comment (cjb) [inline shared/provision/code-page/index.tsx]: s/arg0/code/

File: shared/provision/code-page/qr-scan/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import * as React from 'react'
+export type Props = {
+  waiting: boolean
+  mountKey: string
+  onSubmitTextCode: (arg0: string) => void
```
Comment (cjb) [inline shared/provision/code-page/qr-scan/index.d.ts]: s/arg0/code/

**Pull request merged**